README.md

```diff
@@ -116,44 +115,132 @@ You can additionnaly setup a logger to chose where logs will be output.
+client application should implement to be notified when an operation is request
+by the guest or a network change is detected on the host. It is expected the
+client application will monitor these events to synchronize proxy_wifi with
+other component virtualizing the data path or L3 layer properties.
 
-Proxy_wifi is composed of two main layers: the transport ([Transport.hpp](lib/Transport.hpp)) and the operation handler ([OperationHandler.hpp](lib/OperationHandler.hpp)).
-Implementations of `ProxyWifiService` initialize these two layers.
-The transport is responsible of communications with the proxy_wifi driver on the guest VM, while the operation handler interact with the Windows Wlan API to honor these requests.
+## Proxy_wifi service architecture
 
-### The transport layer
+![Proxy_wifi service architecture](./doc/proxy_wifi_service_architecture.png)
+
+Proxy_wifi is composed of two main layers: the transport
+([Transport.hpp](lib/Transport.hpp)) and the operation handler
+([OperationHandler.hpp](lib/OperationHandler.hpp)).
+
+The concrete implementations of `ProxyWifiService` initialize these two layers
+and expose a clean API to client applications.
+
+The transport is responsible of communications with the proxy_wifi driver on
+the guest VM. The operation handler is responsible of honoring these requests
+and notifying the client application.
+
+### Transport layer
 
 The transport layer communicate with the guest proxy_wifi driver over two channels:
 
-- a request channel handles commands from the proxy_wifi driver and their response. These exchanges are synchronous, only one command is processed at a time.
-- a notification channel handles spontaneous notification from the host, such as network disconnection and signal quality changes.
+- a request channel: it handles commands from the proxy_wifi driver and their
+  responses, such as connection or scan requests.
+- a notification channel: it handles spontaneous notification from the host, such
+  as network disconnection, signal quality changes and scanned bss updates.
+
+The transport layers deals with `Message` ([Messages.hpp](lib/Messages.hpp)) and isn't aware of their content or structure beside the header + body layout.
+
+Messages coming over the request channel are handled as synchronous transactions: once a request is accepted in `Transport::AcceptConnections` ([Transport.hpp](lib/Transport.hpp)), no other request will be handled until the transaction is completed by sending an answer.
 
-The transport layers deals with `Message` ([Messages.hpp](lib/Messages.hpp)) and isn't aware of their content or structure beside the header/body structure.
+On the other hand, messages sent over the notification channel are handled in a fire and forget fashion: the transport send notification asynchronously through a queue and doesn't expect any answer from proxy_wifi driver.
 
 Two implementations of the transport layers are available: over HV sockets or over TCP sockets.
```

Comment:
Is TCP actually still available? I thought we removed that.

Reply:
Since it's only a small amount of code when creating the socket, so we kept it as a potential debugging / test tool.